### PR TITLE
refactor!: remove ZeroCustomFields from ClientState interface

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/02-client/types/legacy_proposal.go [hunk 1]
```diff
@@ -116,15 +115,11 @@ func (up *UpgradeProposal) ValidateBasic() error {
...
-	clientState, err := UnpackClientState(up.UpgradedClientState)
+	_, err := UnpackClientState(up.UpgradedClientState)
 	if err != nil {
 		return errorsmod.Wrap(err, "failed to unpack upgraded client state")
 	}
 
-	if !reflect.DeepEqual(clientState, clientState.ZeroCustomFields()) {
-		return errorsmod.Wrap(ErrInvalidUpgradeProposal, "upgraded client state is not zeroed out")
-	}
-
```
Comment (colin-axner) [inline modules/core/02-client/types/legacy_proposal.go:127]: not necessary, we set the zero custom fields client state regardless

File: modules/light-clients/07-tendermint/client_state.go [hunk 2]
```diff
@@ -174,8 +174,10 @@ func (cs ClientState) Validate() error {
 }
 
 // ZeroCustomFields returns a ClientState that is a copy of the current ClientState
-// with all client customizable fields zeroed out
-func (cs ClientState) ZeroCustomFields() exported.ClientState {
+// with all client customizable fields zeroed out. All chain specific fields must
+// remain unchanged. This client state will used to verify chain upgrades when a
```
Comment (damiannolan) [inline modules/light-clients/07-tendermint/client_state.go:178]:
```suggestion
// remain unchanged. This client state will be used to verify chain upgrades when a
```

**Review by damiannolan — APPROVED** [review]
Clean as a whistle! Love it, happy to see this interface func removed :) 
And nice docs changes too, much appreciated 🙏🏻

File: modules/core/02-client/keeper/keeper.go [hunk 3]
```diff
@@ -439,7 +439,12 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
 // ScheduleIBCSoftwareUpgrade schedules an upgrade for the IBC client.
 func (k Keeper) ScheduleIBCSoftwareUpgrade(ctx sdk.Context, plan upgradetypes.Plan, upgradedClientState exported.ClientState) error {
 	// zero out any custom fields before setting
-	cs := upgradedClientState.ZeroCustomFields()
+	cs, ok := upgradedClientState.(*ibctm.ClientState)
+	if !ok {
+		return errorsmod.Wrapf(types.ErrInvalidClientType, "upgraded client state must be tendermint type, expected: %T, got: %T", &ibctm.ClientState{}, upgradedClientState)
```
Comment (crodriguezvega) [inline modules/core/02-client/keeper/keeper.go:444]: This is usually how we format these error types, and "expected %T" should already give indication that it should be a tendermint type, right?

```suggestion
		return errorsmod.Wrapf(types.ErrInvalidClientType, "expected: %T, got: %T", &ibctm.ClientState{}, upgradedClientState)
```

**Review by crodriguezvega — APPROVED** [review]
Nice one!

**Review by chatton — APPROVED** [review]
LGTM

**Pull request merged**